ml-agents/mlagents/trainers/env_manager.py

```diff
@@ -58,3 +64,36 @@ def get_properties(self) -> Dict[AgentGroup, float]:
     @abstractmethod
     def close(self):
         pass
+
+    def advance(self):
+        # Get new policies if found
+        for brain_name in self.external_brains:
+            try:
+                _policy = self.agent_managers[brain_name].policy_queue.get_nowait()
+                self.set_policy(brain_name, _policy)
+            except AgentManagerQueue.Empty:
+                pass
+        # Step the environment
+        new_step_infos = self._step()
+        # Add to AgentProcessor
+        num_step_infos = self._process_step_infos(new_step_infos)
+        return num_step_infos
+
+    def _process_step_infos(self, step_infos: List[EnvironmentStep]) -> int:
+        for step_info in step_infos:
+            for name_behavior_id in step_info.name_behavior_ids:
+                if name_behavior_id not in self.agent_managers:
+                    logger.warning(
+                        "Agent manager was not created for behavior id {}.".format(
+                            name_behavior_id
+                        )
+                    )
+                    continue
+                self.agent_managers[name_behavior_id].add_experiences(
+                    step_info.current_all_step_result[name_behavior_id],
+                    step_info.worker_id,
+                    step_info.brain_name_to_action_info.get(
+                        name_behavior_id, ActionInfo([], [], {}, [])
```

Comment:
might be worth having a ActionInfo.empty() static method...

Reply:
Added